test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/RequestTests.cs

```diff
@@ -76,6 +71,65 @@ public class RequestTests
             }
         }
 
+        [ConditionalFact]
+        [FrameworkSkipCondition(RuntimeFrameworks.Mono, SkipReason = "Needs to run on 64-bit Mono, which isn't available on OSX.")]
```

Comment:
I think this should be fixed by optimizing the test by either disabling request buffering on the HttpClient or using Sockets directly as suggested below.